internal/dag/executor/command.go

```diff
@@ -40,9 +40,11 @@ func newCommand(ctx context.Context, step dag.Step) (Executor, error) {
 	if len(step.Dir) > 0 && !util.FileExists(step.Dir) {
 		return nil, fmt.Errorf("directory %q does not exist", step.Dir)
 	}
+	stepSpecialExecutionLogPathKey := util.GenerateStepSpecialExecutionLogPathKey(step.NodeID)
 	cmd.Dir = step.Dir
 	cmd.Env = append(cmd.Env, os.Environ()...)
 	cmd.Env = append(cmd.Env, step.Variables...)
+	cmd.Env = append(cmd.Env, fmt.Sprintf("%s=%s", constants.StepDaguExecutionLogPathKeySuffix, os.Getenv(stepSpecialExecutionLogPathKey)))
```

Comment:
Can we change this to something like this?
```suggestion
	// Set environment variables from the DAG context.
	dagContext, err := dag.GetContext(ctx)
	if err != nil {
		return nil, err
	}
	cmd.Env = append(cmd.Env, dagContext.Envs...)
```